pkg/build/builder/daemonless.go

```diff
@@ -272,11 +274,13 @@ func pushDaemonlessImage(sc types.SystemContext, store storage.Store, imageName
 	// }
 	systemContext.AuthFilePath = "/tmp/config.json"
 
-	if authConfig.Username != "" && authConfig.Password != "" {
-		log.V(2).Infof("Setting authentication secret for %q.", authConfig.ServerAddress)
-		systemContext.DockerAuthConfig = &types.DockerAuthConfig{
-			Username: authConfig.Username,
-			Password: authConfig.Password,
+	if len(authConfig) != 0 {
+		for reg, auth := range authConfig {
+			log.V(2).Infof("Setting authentication secret for %q.", reg)
+			systemContext.DockerAuthConfig = &types.DockerAuthConfig{
```

Comment:
(Pushing currently doesn’t push to mirrors anyway, so submitting only one credential via `DockerAuthConfig` is not in itself actually wrong, but it would have to be the credential for the primary registry, not for any of the mirrors. Better to just provide all of them anyway.)

Reply:
rather than even specifying the temp file @mtrmac , perhaps we should just set `systemContext.AuthFilePath` to the location from the search path currently determined in `GetAllDockerAuths` ?

Something akin to the code above that was previously commented out.

Reply:
Yes, sure. I somehow didn’t connect the reading and writing code.

(Do note the caveat about letting the creator of that file (= anyone who can edit the secret object, it seems) specify arbitrary credential helpers.)

Reply:
update on this new direction pushed